[Bug]  the ClassCastException is thrown when creating timestamp data field in spark

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.


### Paimon version

master branch

### Compute Engine

spark-3.3.1

### Minimal reproduce step

CREATE TABLE paimon_dt_decimal (
    uuid int, 
    name string,
    dt timestamp,
    de decimal(5,3)
) TBLPROPERTIES (
    'primary-key' = 'uuid'
);

INSERT INTO paimon_dt_decimal VALUES (1, 'apple', TIMESTAMP'2022-1-2 2:2:2', 3.33);

### What doesn't meet your expectations?

Caused by: java.lang.ClassCastException: java.time.Instant cannot be cast to java.time.LocalDateTime
	at org.apache.paimon.spark.SparkRow.toPaimonTimestamp(SparkRow.java:165)
	at org.apache.paimon.spark.SparkRow.getTimestamp(SparkRow.java:130)
	at org.apache.paimon.table.sink.DynamicBucketRow.getTimestamp(DynamicBucketRow.java:111)
	at org.apache.paimon.data.JoinedRow.getTimestamp(JoinedRow.java:202)
	at org.apache.paimon.data.InternalRow.lambda$createFieldGetter$15c31323$1(InternalRow.java:211)
	at org.apache.paimon.data.InternalRow.lambda$createFieldGetter$781bd14c$1(InternalRow.java:234)
	at org.apache.paimon.data.serializer.InternalRowSerializer.toBinaryRow(InternalRowSerializer.java:158)
	at org.apache.paimon.data.serializer.InternalRowSerializer.serializeToPages(InternalRowSerializer.java:170)
	at org.apache.paimon.data.serializer.InternalRowSerializer.serializeToPages(InternalRowSerializer.java:39)
	at org.apache.paimon.sort.BinaryInMemorySortBuffer.write(BinaryInMemorySortBuffer.java:176)
	at org.apache.paimon.sort.BinaryExternalSortBuffer.write(BinaryExternalSortBuffer.java:123)
	at org.apache.paimon.mergetree.SortBufferWriteBuffer.put(SortBufferWriteBuffer.java:105)
	at org.apache.paimon.mergetree.MergeTreeWriter.write(MergeTreeWriter.java:146)
	at org.apache.paimon.mergetree.MergeTreeWriter.write(MergeTreeWriter.java:53)
	at org.apache.paimon.operation.AbstractFileStoreWrite.write(AbstractFileStoreWrite.java:108)
	at org.apache.paimon.table.sink.TableWriteImpl.writeAndReturn(TableWriteImpl.java:102)
	at org.apache.paimon.table.sink.TableWriteImpl.write(TableWriteImpl.java:97)
	at org.apache.paimon.spark.commands.WriteIntoPaimonTable.$anonfun$run$7(WriteIntoPaimonTable.scala:122)
	at org.apache.paimon.spark.commands.WriteIntoPaimonTable.$anonfun$run$7$adapted(WriteIntoPaimonTable.scala:119)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at org.apache.paimon.spark.commands.WriteIntoPaimonTable.$anonfun$run$6(WriteIntoPaimonTable.scala:119)
	at org.apache.spark.sql.execution.MapPartitionsExec.$anonfun$doExecute$3(objects.scala:201)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:890)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:890)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:136)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/paimon-spark/paimon-spark-2/pom.xml
+++ b/paimon-spark/paimon-spark-2/pom.xml
@@ -80,6 +80,10 @@ under the License.
                     <groupId>org.apache.xbean</groupId>
                     <artifactId>xbean-asm6-shaded</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.glassfish.jersey.media</groupId>
+                    <artifactId>jersey-media-json-jackson</artifactId>
+                </exclusion>
             </exclusions>
         </dependency>
 

--- a/paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkRow.java
+++ b/paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkRow.java
@@ -36,8 +36,10 @@ import org.apache.spark.sql.Row;
 
 import java.io.Serializable;
 import java.sql.Date;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
+import java.time.ZoneId;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
@@ -161,6 +163,10 @@ public class SparkRow implements InternalRow, Serializable {
     private static Timestamp toPaimonTimestamp(Object object) {
         if (object instanceof java.sql.Timestamp) {
             return Timestamp.fromSQLTimestamp((java.sql.Timestamp) object);
+        } else if (object instanceof java.time.Instant) {
+            LocalDateTime localDateTime =
+                    LocalDateTime.ofInstant((Instant) object, ZoneId.systemDefault());
+            return Timestamp.fromLocalDateTime(localDateTime);
         } else {
             return Timestamp.fromLocalDateTime((LocalDateTime) object);
         }

